RealEstateDataCleaning.py

```diff
@@ -8,113 +8,160 @@
 import pandas as pd
 import numpy as np
 
+# aea: argparse here
 input_file = 'Real Estate Model - Candler Park.csv'
 output_file = 'cleaned_data_cp5.csv'
-#lists of keywords to make dummy vars of if they are present in an observation's description  
+
+
+# aea: better as a map features = { 'Exterior' : [...], 'BasementDesc' : [...], ... }
+# aea: this solves capitalization as well (mention PEP8)
+# lists of keywords to make dummy vars of if they are present in an observation's description
 Exterior_features = ['Porch','Patio','Deck','Fenced','Out-Building']
 BasementDesc_features = ['Crawl Space','Slab', 'Basement']
 ParkingDesc_features = ['Driveway','Carport', '1 Car Garage', '2 Car Garage', '3 Car Garage']
 Stories_features = ['1 Story', '2 Story', 'More']
```

Comment:
When we read later in the code, we see that each of these is used in the same way:

```
df = extract_dummies(df, <some list name>, col=<some column name>)
```
This duplicates code which makes things harder to read.  It also introduces a dependency between the "column name" and the "list name" that you have to get correct.  This will be easy to horse up later on the life of this script.  For instance, if we add any new features, we have to modify the script in two different places.

(these variables also have the "problem" that variables are expected to be all lower case, but well fix that here)

We can express the relation between column names and the lists with a simple map.  This then makes our code later very concise:

```
features = {
    'Exterior': ['Porch','Patio','Deck','Fenced','Out-Building'],
    'BasementDesc': ['Crawl Space','Slab', 'Basement'],
    'ParkingDesc': ['Driveway','Carport', '1 Car Garage', '2 Car Garage', '3 Car Garage'],
    'Stories': ['1 Story', '2 Story', 'More'],
}
...
df['BasementDesc'] = df['BasementDesc'].apply(Basement_cleanup)
df['Stories'] = df['Stories'].apply(Stories_cleanup)

for column, items in features.items():
    df = extract_dummies(df, items, col=column)
```

Now to add a new list of features, we only modify this map and we get it for free:

```
features = {
   "GameRoomDesc" : ["Big TV", "Hottub", "X Box"],
   ...
}
```